### ISSUE TITLE
Rework `datadog_metrics` sink to avoid cloning metadata

### ISSUE THREAD

**Issue by bruceg** [vectordotdev/vector#7637]
The initial acknowledgement support in the `datadog_metrics` sink was done in the simplest way by cloning the metadata. This is due to the nature of how metric events flow through the pipe and batch buffers. This is less than ideal and should be reworked to avoid the clone. However, this will likely affect several metrics-related interface signatures and so is more than a trivial fix, and it will be best to happen after the metrics batch buffer rework.

Ref #7535
Blocked by #6045

**Comment by bruceg** [comment]
This was worked around by the `take_finalizers` work in #8426 and so is no longer relevant.
